js/adapt-stateful-session.js

```diff
@@ -127,6 +145,16 @@ define([
 			Adapt.offlineStorage.set("interaction", id, response, result, latency, responseType);
 		},
 
+		onResetSession: function () {
+			this.removeEventListeners();
+			this.setupEventListeners();
+			var sessionPairs = this.getSessionState();
+			
+			// hard status reset
+			Adapt.offlineStorage.set("status", "incomplete");
+			Adapt.offlineStorage.set(sessionPairs);
```

Comment:
when you reload the course json, it creates a new set of collections. This resets all the progress in the Model Data. That's why I also reset status to incomplete.

What I know by hard:
It will keep the Assessment object in offlinestorage. This can be used to restore session store for each language during multiple language switches.
If a Score is available, it will remain untio it is overwritten with new score.
Not sure about question interaction, will need to check